Move colors definitions from the components less files to default.less

Nouzbe opened the pull request:
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
^ I did not manage to get the Linter to work :/
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

----

**Related issue**: #11519 

**Goal**: Move all colors definitions from the components less indexes into `default.less` so that the whole style can be controlled with less variables. Specifically, it makes it easier (and less hacky) to create dark themed applications using Ant Design.

**Note**: New variables were introduced. This is just a proposal and the way I factorized colors and named variables can be discussed.

**Possible next step**: Adding a custom Lint rule to check that colors are not added to components less indexes in the future would make this more robust.

Nouzbe committed:
Move colors definitions from the components less files to default.less

Nouzbe committed:
Changes after review
	- renamed @badge-color into @badge-text-color
	- renamed @alternate-text-color into @text-color-inverse
	- button/style/mixin.less: changed 'transparent' back to @component-background (value is #fff by default)

afc163 commented on components/alert/style/index.less:
```diff
@@ -74,7 +74,7 @@
       transition: color .3s;
       &:hover {
-        color: #404040;
+        color: @icon-color;
```
@icon-color-hover;

afc163 commented on components/spin/style/index.less:
```diff
@@ -40,7 +40,7 @@
         width: 100%;
         padding-top: (@spin-dot-size - @font-size-base) / 2 + 2px;
-        text-shadow: 0 1px 2px #fff;
+        text-shadow: 0 1px 2px @alternate-shadow-color;
```
`@shadow-color-inverse`, save rule as `@text-color-inverse`

afc163 commented on components/style/themes/default.less:
```diff
@@ -12,6 +12,8 @@
 @highlight-color        : @red-6;
 @warning-color          : @gold-6;
+@default-color          : #fff;
+@alternate-color        : #000;
```
I don't really understand what `alternate` stands for which is sometimes white and sometimes black.

Nouzbe replied:
Maybe `default-light-color` and `default-dark-color` would be better, or even `@black` and `@white`.

I added these two in order to be able to replace the harcoded shades of grey by `shade(@black, x%)`, which has the benefit of working with any underlying background (as opposed to `#444` for instance, which assumes that the background is white and thus does not work if `@component-background` is assigned another value).

afc163 replied:
`@black` & `@white` is nice and simple enough.

Nouzbe committed:
Changes after review
	- renamed @default-color into @white
	- renamed @alternate-color into @black
	- renamed @alternate-shadow-color into @shadow-color-inverse
	- renamed @icon-color into @icon-color-hover

afc163 commented on components/style/themes/default.less:
```diff
@@ -93,6 +97,7 @@
 // Border color
 @border-color-base      : hsv(0, 0, 85%);  // base border outline a component
 @border-color-split     : hsv(0, 0, 91%);  // split border inside a component
+@alternate-border-color : @white;
```
@border-color-inverse

afc163 commented on components/style/themes/default.less:
```diff
@@ -110,7 +115,8 @@
 // Shadow
-@shadow-color           : rgba(0, 0, 0, .15);
+@shadow-color           : fade(@black, 15%);
+@shadow-color-inverse : @component-background;
```
It is better to align colon here.

afc163 commented on components/input-number/style/index.less:
```diff
@@ -26,7 +26,7 @@
     font-weight: bold;
     &:active {
-      background: #f4f4f4;
+      background: @input-active-bg;
```
`@input-number-handler-active-bg`

afc163 commented on components/style/themes/default.less:
```diff
@@ -386,7 +394,8 @@
 @card-actions-background: @background-color-light;
-@card-shadow: 0 2px 8px rgba(0, 0, 0, .09);
+@card-shadow: 0 2px 8px fade(@black, 9%);
+@card-background: #cfd8dc;
```
I can't find `@card-background`'s usage.

Nouzbe replied:
It is used in [components/card/style/index.less#L8](https://github.com/Nouzbe/ant-design/blob/feature-3.8.0/components/card/style/index.less#L8).

Nouzbe committed:
Changes after review
	- renamed @alternate-border-color into @border-color-inverse (was the last "alernate" color)
	- fixed default.less alignment
	- renamed @input-active-bg into @input-number-handler-active-bg

Nouzbe commented:
@afc163 does it make it easier to you if I rebase with `feature-3.9.0`?

Nouzbe commented:
Let's start fresh in #11779 (squashed changes and rebased `feature-3.9.0`)
